File: src/test/java/configsources/JsonConfigSourceTest.java
package configsources;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import uk.dsx.reactiveconfig.PropertyTypesKt;
import uk.dsx.reactiveconfig.ReactiveConfig;
import uk.dsx.reactiveconfig.Reloadable;
import uk.dsx.reactiveconfig.configsources.JsonConfigSource;
import uk.dsx.reactiveconfig.interfaces.ConfigSource;

import java.io.File;
import java.nio.file.Paths;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;

public class JsonConfigSourceTest {
    private ConfigSource jsonSource;
    private ReactiveConfig config;
    @BeforeEach
    void init(){
        jsonSource =
                new JsonConfigSource(Paths.get("src" + File.separator + "test" + File.separator + "resources"), "jsonSource.json");

        config = new ReactiveConfig.Builder()
                .addSource("jsonConfig", jsonSource)
                .build();
    }

    @Test
    void booleanTypeTest(){
        Reloadable<Boolean> isSomethingOn = config.get("isSomethingOn", PropertyTypesKt.booleanType);
        assertTrue(isSomethingOn.get());
    }

    @Test
    void stringTypeTest(){
        Reloadable<String> property = config.get("property", PropertyTypesKt.stringType);
        assertEquals("someInfo", property.get());
    }
}

File: src/test/kotlin/uk/dsx/reactiveconfig/configsources/JsonConfigSourceTest.kt
package uk.dsx.reactiveconfig.configsources

import org.spekframework.spek2.Spek
import org.spekframework.spek2.style.specification.describe
import uk.dsx.reactiveconfig.*
import java.io.File
import java.nio.file.Paths
import kotlin.test.assertEquals
import kotlin.test.assertNotNull
import kotlin.test.assertTrue

object JsonConfigSourceTest : Spek({
    val jsonSource =
        JsonConfigSource(Paths.get("src" + File.separator + "test" + File.separator + "resources"), "jsonSource.json")

    val config = ReactiveConfig.Builder()
        .addSource("jsonConfig", jsonSource)
        .build()

    describe("resource check") {
        it("should be initialised") {
            assertNotNull(jsonSource)
        }
    }

    describe("checks reading properly BooleanNode from json") {
        val isSomethingOn = config.getReloadable("isSomethingOn", booleanType)

        it("should contain value 'true' sent from JsonConfigSource") {
            assertTrue(isSomethingOn!!.get())
        }
    }

    describe("checks reading properly StringNode from json") {
        val property = config.getReloadable("property", stringType)

        it("should contain value 'someInfo' sent from JsonConfigSource") {
            assertEquals("someInfo", property!!.get())
        }
    }
})